### ISSUE TITLE
PrefixAllGlobals/OverrideGlobals: detect non-prefixed variables when created in control structures

### ISSUE THREAD

**Issue by jrfnl** [WordPress-Coding-Standards/WordPress-Coding-Standards#1236]
The declaration of the new `$something` variable in code like the below is currently not detected by the `PrefixAllGlobals`, nor the `GlobalVariables` sniff while it does create a new non-prefixed variable and could easily dangerously override a WP native global.

```php
$prefix_something = array(...);
foreach ( $prefix_something as $something ) {
    // Do something.
}

while ( ( $something = function_call() ) === true ) {
	// Do something.
}
```

Along the same lines, but generally less problematic, the `$i` variable declaration in the code below is also not detected.
```php
for ( $i = 0; $i < 10; $i++ ) {
	// Do something.
}
```

Of course, this only applies when these constructs are placed directly in the global namespace.

**Comment by JDGrimes** [comment]
Yes, I noticed this a while back and didn't end up getting an issue created. I believe that it is because of the check for whether or not the variable is within a condition of any kind. I guess it is intended to be directed at functions, but ends up picking up conditions as well.

**Comment by jrfnl** [comment]
Just been running some tests. Looks like that from the examples above, it's only the `foreach` which is not being picked up.
The `while` and `for` both report an error already (providing they are in the global namespace).
